transport/peer/x/peerlist/roundrobin/roundrobin.go

```diff
@@ -65,9 +71,12 @@ func (pl *RoundRobin) addAll(peerIDs []transport.PeerIdentifier) error {
 
 // Add a peer identifier to the round robin
 func (pl *RoundRobin) Add(pid transport.PeerIdentifier) error {
+	pl.lock.Lock()
+	defer pl.lock.Unlock()
```

Comment:
If there is a single exit path, don't defer. There's some cost associated with
that operation.

```
pl.lock.Lock()
err := pl.addPeer(pid)
pl.lock.Unlock()
return err
```